### PR TITLE
Build fixes and improvements

### PR DESCRIPTION
**Pull request by podcastinator**

### CODE REVIEW

File: demo/c-api/c-api-demo.cc [hunk 1]
```diff
@@ -151,13 +156,13 @@ int main(int argc, char** argv) {
   }
   
   // save model
-  const char* fname = "/home/xgb/secure-xgboost/demo/c-api/demo_model.model";
-  safe_xgboost(XGBoosterSaveModel(booster, fname));
+	std::string fname(cwd + "/demo_model.model");
```
Comment (chester-leung) [inline demo/c-api/c-api-demo.cc]: Small nit: this line looks strangely indented in my editor. Is it the same for you too?

**Review by chester-leung — COMMENTED** [review]
Just one formatting thing on line 159 in `c-api-demo.cc`

File: demo/c-api/c-api-demo.cc (hunk 1 above)
Comment (podcastinator) [reply demo/c-api/c-api-demo.cc]: Yeah, weird. Amended the commit. We should start using the formatter.

**Comment by podcastinator** [comment]
Amended the commit. We should start using the formatter :P

**Review by chester-leung — APPROVED** [review]

**Comment by chester-leung** [comment]
Thanks!

**Pull request merged**